reVX/plexos/plexos_plants.py

```diff
@@ -0,0 +1,1006 @@
+        dists : list
+            List of distances from plants to sc_gids
+        bus_dists : list
+            List of distances from bus associated with plants to sc_gids
+        """
+        unique_gids, plant_gids = np.unique(sc_gids, return_inverse=True)
+        for i, sc_gid in enumerate(unique_gids):
+            sc_gid = int(sc_gid)
+            if self.sc_points.check_sc_gid(sc_gid):
+                plant_ids = np.where(plant_gids == i)[0]
+                if len(plant_ids) > 1:
+                    sc_dists = dists[plant_ids]
+                    if len(sc_dists) != len(np.unique(sc_dists)):
+                        idxs = np.argsort(bus_dists[plant_ids])
+                    else:
+                        idxs = np.argsort(sc_dists)
+
+                    plant_ids = plant_ids[idxs]
+
+                for plant_id in plant_ids:
+                    capacity = self.plant_capacity[plant_id]
+                    if (capacity > 0) and self.sc_points.check_sc_gid(sc_gid):
+                        sc_point, sc_capacity = \
+                            self.sc_points.get_capacity(sc_gid, capacity)
+                        if sc_capacity:
+                            plant = self[plant_id]
+                            if plant is None:
+                                plant = [sc_point]
+                            else:
+                                plant.append(sc_point)
+
+                            self[plant_id] = plant
+                            self._capacity[plant_id] -= sc_capacity
+                            logger.debug('Allocating {:.1f}MW to plant {} from'
+                                         ' sc_gid {}'.format(sc_capacity,
+                                                             plant_id,
+                                                             sc_gid))
+                        else:
+                            msg = ('WARNING: sc_gid {} returned 0 capacity!'
+                                   .format(sc_gid))
+                            logger.warning(msg)
+                            warn(msg)
+
+    def _fill_plants(self, plants):
+        """
+        Fill plants with capacity from supply curve points
+
+        Parameters
+        ----------
+        dist_percentile : int, optional
+            Percentile to use to compute distance threshold using sc_gid to
+            SubStation distance , by default 90
+        lcoe_col : str, optional
+            LCOE column to sort by, by default 'total_lcoe'
+        lcoe_thresh : float, optional
+            LCOE threshold multiplier, exclude sc_gids above threshold,
+            by default 1.3
+        max_workers : int, optional
+            Number of workers to use for plant sc extraction, 1 == serial,
+            > 1 == parallel, None == parallel using all available cpus,
+            by default None
+
+        Returns
+        -------
+        plants : list
+            List of plant supply curve tables
```

Comment:
Is this actually supposed to return something? Doesn't look like it does

Reply:
ditto